crates/flashblocks-rpc/src/rpc.rs

```diff
@@ -419,179 +207,87 @@ where
-                }
-            }
-        } else {
-            // Handle cache lookup for transactions not found in the main lookup
-            if let Some(tx) = self
-                .cache
-                .get::<OpTransactionSigned>(&CacheKey::Transaction(tx_hash))
-            {
-                let block_number = self
-                    .cache
-                    .get::<u64>(&CacheKey::TransactionBlockNumber(tx_hash))
-                    .unwrap();
-                let block = self
-                    .cache
-                    .get::<OpBlock>(&CacheKey::Block(block_number))
-                    .unwrap();
-                let index = self
-                    .cache
-                    .get::<u64>(&CacheKey::TransactionIndex(tx_hash))
-                    .unwrap();
-                let tx_info = TransactionInfo {
-                    hash: Some(tx.tx_hash()),
-                    block_hash: Some(block.header.hash_slow()),
-                    block_number: Some(block.number),
-                    index: Some(index),
-                    base_fee: block.base_fee_per_gas,
-                };
-                let tx = get_recovered_tx(&self.cache, tx_hash);
-                Ok(Some(self.transform_tx(tx, tx_info, None)))
-            } else {
-                Ok(None)
-            }
+        if let Some(fb_transaction) = self.flashblocks_state.get_transaction_by_hash(tx_hash) {
+            self.metrics.get_transaction_receipt.increment(1);
+            return Ok(Some(fb_transaction));
         }
+
+        Ok(EthTransactions::transaction_by_hash(&self.eth_api, tx_hash)
+            .await?
+            .map(|tx| tx.into_transaction(self.eth_api.tx_resp_builder()))
+            .transpose()?)
     }
 
     async fn send_raw_transaction_sync(
         &self,
         transaction: alloy_primitives::Bytes,
-    ) -> RpcResult<Option<RpcReceipt<Optimism>>> {
+    ) -> RpcResult<RpcReceipt<Optimism>> {
+        debug!(message = "rpc::send_raw_transaction_sync");
+
         let tx_hash = match EthTransactions::send_raw_transaction(&self.eth_api, transaction).await
         {
             Ok(hash) => hash,
             Err(e) => return Err(e.into()),
         };
 
         debug!(
-            message = "sent raw transaction successfully",
+            message = "rpc::send_raw_transaction_sync::sent_transaction",
             tx_hash = %tx_hash
         );
 
-        match self.wait_for_receipt(tx_hash).await {
-            Some(receipt) => {
-                debug!(
-                    message = "received receipt for transaction",
-                    tx_hash = %tx_hash
-                );
-                Ok(Some(receipt))
-            }
-            None => {
-                debug!(
-                    message = "timeout waiting for receipt",
-                    tx_hash = %tx_hash
-                );
-                Ok(None)
+        const TIMEOUT_DURATION: Duration = Duration::from_secs(6);
```

Comment:
I wonder if this should be client-side configurable, with a reasonable max limit so we don't enter a deadlock

Reply:
Unfortunately the API doesn't have this, my 2c is lets start w/ 3 blocks and if there's a demand for more configuration we can setup a flag for it